src/inmanta/server/services/projectservice.py

```diff
@@ -110,8 +109,11 @@ async def project_delete(self, project_id: uuid.UUID) -> None:
             raise NotFound("The project with given id does not exist.")
 
         environments = await data.Environment.get_list(project=project.id)
-        for env in environments:
-            await asyncio.gather(self.autostarted_agent_manager.stop_agents(env, delete_venv=True), env.delete_cascade())
+        if len(environments) > 0:
+            raise ServerError(
```

Comment:
A `ServerError` is an HTTP 500, so an unexpected exception on the server. This should be ` Conflict` (409), because the current state of the resource doesn't allow the action.